luxonis_train/attached_modules/visualizers/classification_visualizer.py

```diff
@@ -33,17 +34,28 @@ def __init__(
         self.font_scale = font_scale
         self.color = color
         self.thickness = thickness
+        self.multilabel = multilabel
 
     def _get_class_name(self, pred: Tensor) -> str:
-        idx = int((pred.argmax()).item())
-        if self.class_names is None:
-            return str(idx)
-        return self.class_names[idx]
+        """Handles both single-label and multi-label classification."""
+        if self.multilabel:
+            idxs = (pred > 0.5).nonzero(as_tuple=True)[0].tolist()
+            if self.class_names is None:
+                return ", ".join([str(idx) for idx in idxs])
+            return ", ".join([self.class_names[idx] for idx in idxs])
+        else:
+            idx = int((pred.argmax()).item())
+            if self.class_names is None:
+                return str(idx)
+            return self.class_names[idx]
 
     def _generate_plot(
         self, prediction: Tensor, width: int, height: int
     ) -> Tensor:
-        pred = prediction.softmax(-1).detach().cpu().numpy()
+        if self.multilabel:
+            pred = prediction.sigmoid().detach().cpu().numpy()
+        else:
```

Comment:
Yes, binary classification is handled correctly via `softmax `for single-label classification.

Reply:
But with binary classification (output is 1 value) the predicted confidence is not correct. Let's say the NN predicts value=2. If we use sigmoid this will mean that it is ~88% confident (after `prediction.sigmoid()`) it is this class while with softmax whatever the value the confidence will always be 1 (after `prediction.softmax()`)

Reply:
In our binary classification task, the output of the network is a vector of length 2 ([here](https://github.com/luxonis/luxonis-train/blob/6146da6422388ed97945d42bcd1c4625c2bc9d68/luxonis_train/nodes/heads/classification_head.py#L29)). Therefore, using softmax to normalize this vector is correct, as it will produce the proper confidence values for each class.